src/main/java/org/jabref/model/groups/WordKeywordGroup.java

```diff
@@ -148,4 +136,59 @@ public int hashCode() {
                 keywordSeparator,
                 onlySplitWordsAtSeparator);
     }
+
+    interface SearchStrategy {
+        boolean contains(BibEntry entry);
+    }
+
+    class StringSearchStrategy implements SearchStrategy {
```

Comment:
I would put the class in a seperate `.java` class. This would also enable testing with `...Test.java` files.

Reply:
This is more an implementation detail which shouldn't exposed to the outside world in my opinion. Moreover, private variables are used which makes extraction awkward.